pkg/controller/endpointslice/utils_test.go

```diff
@@ -223,9 +268,12 @@ func TestPodToEndpoint(t *testing.T) {
 			node: node1,
 			svc:  &svc,
 			expectedEndpoint: discovery.Endpoint{
-				Addresses:  []string{"1.2.3.5"},
-				Conditions: discovery.EndpointConditions{Ready: utilpointer.BoolPtr(true)},
-				Hostname:   &readyPodHostname.Spec.Hostname,
+				Addresses: []string{"1.2.3.5"},
+				Conditions: discovery.EndpointConditions{
+					Ready:       utilpointer.BoolPtr(true),
+					Terminating: utilpointer.BoolPtr(false),
```

Comment:
Are there any instances where we would expect terminating to be nil when the feature gate is enabled? It looks like all of these tests expect true or false values.

Reply:
I don't think there is, at least in the context of endpointslice controller that is only adding pod endpoints. The terminating state would be either `true` or `false` depending on whether `DeletionTimestamp` is nil. Maybe I missed an edge case here though :thinking: 

Will add some tests for when the gate is disabled though